backend/poll.go

```diff
@@ -0,0 +1,184 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"golang.org/x/net/context"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/datastore"
+	"net/http"
+)
+
+const (
+	ERROR_CASE_POLL  = "error"
+	POLL_SAMPLE_PATH = "/" + CATEGORY_SAMPLE_TEMPLATES + "/poll/"
+)
+
+type PollForm struct {
+	ClientId string
+	Answer   string
+}
+
+type Result struct {
+	Percentage []int
+	Answer     string
+}
+
+type Answer struct {
+	AnswerText string
+	Votes      int
+}
+
+func InitPollSample() {
+	http.HandleFunc(POLL_SAMPLE_PATH+"submit-xhr", submitXHR)
+}
+
+func getAnswersFromDatastore(context context.Context) ([]Answer, error) {
+	query := datastore.NewQuery("Answer")
```

Comment:
Extract `"Answer"` into a constant - you use the same string at least three times.